onnxruntime/python/onnxruntime_pybind_ortvalue.cc

```diff
@@ -210,10 +210,18 @@ void addOrtValueMethods(pybind11::module& m) {
 #ifdef ENABLE_TRAINING
       .def("to_dlpack", [](OrtValue* ort_value) -> py::object {
         return py::reinterpret_steal<py::object>(ToDlpack(*ort_value));
-      })
+      }, "Returns a DLPack representing the tensor.")
       .def_static("from_dlpack", [](py::object data, bool is_bool_tensor = false) {
         return FromDlpack(data.ptr(), is_bool_tensor);
-      })
+      }, "Convers a tensor from a external library into an OrtValue by means of the __dlpack__ protocol.")
+      .def("__dlpack__", [](OrtValue* ort_value) -> py::object {
```

Comment:
PyTorch and some other platforms (such as Numpy)'s __dlpack__ also has a 'stream' argument, if it's CUDA/ROCM device, we need to sync the stream if streams are not same. Maybe we can skip such logic for now as we think that ORT and PT are running in same default stream, but maybe we still need to keep that argument here?

Note that PyTorch support such __dlpack__ and __dlpack_device__ from 1.10, currently we mainly run our ORTModule on 1.9.1, so this won't help if we use old PT version.

Reply:
I looked into that, it is current written in python: https://github.com/pytorch/pytorch/blob/master/torch/_tensor.py#L1113. I need to move it to C++. We should keep the argument. It is not used now as this protocol is only used between ort and pytorch.

Reply:
pytroch code about stream:

```
        if stream is not None and type(stream) is not int:
            # Stream pointers in CUDA/ROCm are uniquely numbered and can
            # be retrieved from their integer value.
            raise TypeError('stream must be ``int`` or ``none``')
        elif stream is not None and stream != -1:
            if self.device.type == 'cuda':
                stream = torch.cuda.streams.ExternalStream(stream)
                # Only synchronize on different streams
                if stream != torch.cuda.current_stream:
                    event = torch.cuda.Event()
                    event.record(torch.cuda.current_stream())
                    stream.wait_event(event)
```